sycl/doc/SYCLCompilerAndRuntimeDesign.md

```diff
@@ -395,7 +395,35 @@ llvm-no-spir-kernel host.bc
 
 It returns 0 if no kernels are present and 1 otherwise.
 
-
+#### Device code splitting
+
+Putting all device code into a single SPIRV module does not work well in the
+following cases:
+1. There are thousands of kernels defined and only small part of them is used at
+run-time. Having them all in one SPIR-V module significantly increases JIT time.
+2. Device code can be specialized for different devices. For example, kernels
+which are supposed to be executed only on CPU can contain standard C library calls
```

Comment:
```suggestion
that are supposed to be executed only on FPGA can use extensions available for FPGA only.
```
Using host function sounds like specification violation unless it's extension.

Reply:
Changed, thanks